### ISSUE TITLE
Optimizing battery usage kills widget update

### ISSUE THREAD

**Issue by infinitewaveparticle** [thuryn/your-local-weather#100]
If the Android option to optimize battery usage is turned off the widget automatically updates, but when it's turned on the widget only updates manually.

**Comment by thuryn** [comment]
I don't believe there is a way how to run updates with battery optimization - Android simply kills everything and the updates does not work then. I'll try to look at this issue some time in the future.

**Comment by infinitewaveparticle** [comment]
Take a look at the source code for the following app:

https://github.com/hwki/SimpleBitcoinWidget

I use it, and the app is Optimized. It updates just fine.

**Comment by thuryn** [comment]
I'll check the project you've send and try to do my best to find what part off app needs battery optimization switch on

**Comment by infinitewaveparticle** [comment]
@thuryn Is this fixed? Also, do I need to use a wake-up strategy? I currently do not. I'm a battery saver... my phone is a Galaxy S7... used it since 2016... my phone stays on for months at a time and drops only about 1% per hour when not in use whether on Wi-Fi or cellular.

**Comment by thuryn** [comment]
There are two things to consider. First is battery optimization - I'm not able to switch batter optimization off on my phone and therefore I'm not able to check if the app keeps running then the optimization is off. The second is wake-up strategy - If you don't need to wake the phone you are a lucky man. I've got Honor 8 phone - the phone needs to have display on to get correct network information (otherwise it reports location where I was 1 hour before).
